thorlcr/activities/countproject/thcountprojectslave.cpp

```diff
@@ -102,7 +102,7 @@ class LocalCountProjectActivity : public BaseCountProjectActivity
 class CountProjectActivity : public BaseCountProjectActivity, implements ILookAheadStopNotify
 {
     typedef BaseCountProjectActivity PARENT;
-    bool first = false; // until start
+    std::atomic<bool> first; // false until start
```

Comment:
I don't see why this should be atomic.
'first' is used in start()/nextRow()/stop() - which should all be thread safe. Almost always called by a single thread, or in unusual circumstances, control passed over to another.